tf_files/aws/modules/eks-nodepool/cloud.tf

```diff
@@ -0,0 +1,342 @@
+  type                     = "ingress"
+  from_port                = 443
+  to_port                  = 443
+  protocol                 = "tcp"
+  #security_group_id        = "${aws_security_group.eks_control_plane_sg.id}"
+  security_group_id        = "${var.control_plane_sg}"
+  source_security_group_id = "${aws_security_group.eks_nodes_sg.id}"
+  #depends_on               = ["aws_security_group.eks_nodes_sg", "aws_security_group.eks_control_plane_sg" ]
+  depends_on               = ["aws_security_group.eks_nodes_sg"]
+}
+
+
+resource "aws_security_group_rule" "communication_plane_to_nodes" {
+  type                     = "ingress"
+  from_port                = 1025
+  to_port                  = 65534
+  protocol                 = "tcp"
+  security_group_id        = "${aws_security_group.eks_nodes_sg.id}"
+  #source_security_group_id = "${aws_security_group.eks_control_plane_sg.id}"
+  source_security_group_id = "${var.control_plane_sg}"
+  #depends_on               = ["aws_security_group.eks_nodes_sg", "aws_security_group.eks_control_plane_sg" ]
+  depends_on               = ["aws_security_group.eks_nodes_sg"]
+}
+
+resource "aws_security_group_rule" "nodes_internode_communications" {
+  type = "ingress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  description       = "allow nodes to communicate with each other"
+  security_group_id = "${aws_security_group.eks_nodes_sg.id}"
+  self              = true
+}
+
+resource "aws_security_group_rule" "nodes_interpool_communications" {
+  type = "ingress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  description       = "allow default nodes to communicate with each other"
+  security_group_id = "${aws_security_group.eks_nodes_sg.id}"
+  source_security_group_id = "${var.default_nodepool_sg}"
+}
+
+## Worker Node AutoScaling Group
+# Now we have everything in place to create and manage EC2 instances that will serve as our worker nodes
+# in the Kubernetes cluster. This setup utilizes an EC2 AutoScaling Group (ASG) rather than manually working with
+# EC2 instances. This offers flexibility to scale up and down the worker nodes on demand when used in conjunction
+# with AutoScaling policies (not implemented here).
+
+
+resource "aws_launch_configuration" "eks_launch_configuration" {
+  associate_public_ip_address = false
+  iam_instance_profile        = "${aws_iam_instance_profile.eks_node_instance_profile.name}"
+  image_id                    = "${data.aws_ami.eks_worker.id}"
+  instance_type               = "${var.instance_type}"
+  name_prefix                 = "eks-${var.vpc_name}-nodepool-${var.nodepool}"
+  security_groups             = ["${aws_security_group.eks_nodes_sg.id}", "${aws_security_group.ssh.id}"]
+  user_data_base64            = "${base64encode(data.template_file.bootstrap.rendered)}"
+  key_name                    = "${var.ec2_keyname}"
+
+  root_block_device {
+    volume_size = 30
+  }
```

Comment:
We should check with Zac or Phillis about this.  I'm not sure how the `jupyter-fuse` stuff works exactly - if an app starts reading an S3 object, does the whole object get copied down locally, or does it just copy down a block at a time on the fly, and never actually stores the whole object locally?